js/bybit.js

```diff
@@ -1617,8 +1674,30 @@ module.exports = class bybit extends Exchange {
         let intervalKey = 'interval';
         let sinceKey = 'from';
         const isUsdcSettled = market['settle'] === 'USDC';
+        const enableUnifiedMargin = this.safeValue (this.options, 'enableUnifiedMargin');
```

Comment:
with this unified margin integration, we have huge methods with a lot of endpoints logic, request params, etc, making them very hard to maintain (we never know if we might break something accidentally), and even more important to remove the v2 logic after bybit deprecates it. 

With that in mind, I would like to suggest this approach, for every unified method we would split the v2 and v3 logic into different methods, something like this:

```Javascript
    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
        // common logic (parse timeframe, load market, etc)
        if (spot) {
            response = this.fetchSpotOHLCV (symbol, timeframe, since, limit, params);
        }
        if (unifiedMargin) {
            response = this.fetchUnifiedMargintOHLCV (symbol, timeframe, since, limit, params);
        } else {
            response = this.fetchDerivativesOHLCV (symbol, timeframe, since, limit, params);
        }
    }
```

In the future, to deprecate v2 we just need to comment out one line and remove the method with the isolated v2 logic, which is much easier than dissecting the current logic months after implementing it.

what do you think? @sc0Vu @kroitor

Reply:
@carlosmiei they did change requests payload and response, for now I can only test manually.

For the future upgrading, I agree with you, it's might be easier if we separate. But if we do that, isn't it over-engineering? (There will be some duplicate code blocks.)

Reply:
@sc0Vu I know that we would get some duplication the thing is I suspect that v3 and v2 logic are getting too tied and will be hard to safely split in the future or even make to apply some fix if needed, @kroitor what do you think?

Reply:
I would totally agree with the approach proposed by @carlosmiei ! I think in general we should refrain from mixing up the logic, ideally just split them into exchange-specific methods and call those methods under an if-else conditional, as shown above.

**Sidenote**: I'm not sure how margin OHLCV are different from regular spot OHLCV with bybit v2 vs v3, they're usually the same thing with the rest of the exchanges – margin trading is done with the same set of pairs as regular spot trading, the only difference is the underlying settlement logic within the exchange engine. Or, put simply, spot trading is the same as margin trading with leverage 1 without borrowing. Let me know if I'm missing something here.